core/dbt/task/runnable.py

```diff
@@ -360,21 +357,25 @@ def execute_nodes(self):
         pool = ThreadPool(num_threads)
         try:
             self.run_queue(pool)
-
         except FailFastError as failure:
             self._cancel_connections(pool)
+
+            node_results_ids = [r.node.unique_id for r in self.node_results]
+
+            for r in self._flattened_nodes:
+                if r.unique_id not in node_results_ids:
+                    self.node_results.append(skip_result(r, "Skipping due to fail_fast"))
+
             print_run_result_error(failure.result)
             raise
-
         except KeyboardInterrupt:
             self._cancel_connections(pool)
             print_run_end_messages(self.node_results, keyboard_interrupt=True)
             raise
-
-        pool.close()
-        pool.join()
-
-        return self.node_results
+        finally:
+            pool.close()
+            pool.join()
+            return self.node_results
```

Comment:
Out of curiosity, why return is also moved into the `finally` block?

Reply:
But doesn't it mean we no longer return `self.node_results` when no exception happens?

Reply:
I believe `finally` blocks are executed whether an exception is raised or not